### ISSUE TITLE
🐛 HTML parser rejects `\u` literal in attribute values

### ISSUE THREAD

**Issue by aplefull** [biomejs/biome#10513]
### Environment information

<details>

```bash

CLI:
  Version:                      2.4.16
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_PATH:                    unset
  BIOME_LOG_PREFIX_NAME:             unset
  BIOME_LOG_LEVEL:                   unset
  BIOME_LOG_KIND:                    unset
  BIOME_CONFIG_PATH:                 unset
  BIOME_THREADS:                     unset
  BIOME_WATCHER_KIND:                unset
  BIOME_WATCHER_POLLING_INTERVAL:    unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v26.2.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         pnpm/11.5.0

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false
  HTML full support enabled:    true

Workspace:
  Open Documents:               0

```

</details>

### What happened?

Biome reports parse errors on a valid HTML file when an attribute value contains a literal backslash followed by u:
```html
<div data-test="\u"></div>
```

[Playground url](https://biomejs.dev/playground/?tab=formatter&pane=Diagnostics&code=PABkAGkAdgAgAGQAYQB0AGEALQB0AGUAcwB0AD0AIgBcAHUAIgA%2BADwALwBkAGkAdgA%2BAA%3D%3D&language=html)

`biome lint` result:
```bash
report.html:1:16 parse ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Expected an initializer but instead found '"\u"'.
  
  > 1 │ <div data-test="\u"></div>
      │                ^^^^
  
  ℹ Expected an initializer here.
  
  > 1 │ <div data-test="\u"></div>
      │                ^^^^
  

report.html:1:17 parse ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Invalid unicode sequence
  
  > 1 │ <div data-test="\u"></div>
      │                 ^^
  
  ℹ Non hexadecimal number
  
  > 1 │ <div data-test="\u"></div>
      │                   ^
  
  ℹ A unicode escape sequence must consist of 4 hexadecimal numbers: `\uXXXX`, e.g. `\u002F' for '/'.
  

report.html:1:21 parse ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Expected an attribute but instead found '<'.
  
  > 1 │ <div data-test="\u"></div>
      │                     ^
  
  ℹ Expected an attribute here.
  
  > 1 │ <div data-test="\u"></div>
      │                     ^
  

Checked 12 files in 27ms. No fixes applied.
Found 3 errors.
```

### Expected result

Biome should parse and format the file without errors.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
I'm inclined to say this is intentional behavior. Like the diagnostics indicate, `\u` is the start of a unicode escape sequence.

What does the HTML spec say about this?

**Comment by aplefull** [comment]
Maybe you are thinking about JavaScript escapes? There is no backslash escape mechanism in the HTML tokenizer. 

Paragraph [13.2.5.36](https://html.spec.whatwg.org/multipage/parsing.html#attribute-value-(double-quoted)-state) defines the "Attribute value (double-quoted) state" with exactly four cases:
`U+0022 QUOTATION MARK (")`, `U+0026 AMPERSAND (&)`, `U+0000 NULL`, `EOF`. Anything else must be appended to the current attribute's value. 

The only escape mechanism HTML defines for attributes is triggered by `&`, like `&amp;`. So `<div data-test="\u"></div>` is valid HTML. For example this weird code works just fine, because `\` is taken literally:
```html
<div class="\u"></div> 
<style>
.\\u {
  width: 100px;
  height: 100px;
  background-color: red;
}
</style>   
```

**Comment by dyc3** [comment]
Thank you, that's the context I needed.
